FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ExploreFragment.java

```diff
@@ -105,37 +136,108 @@ public void onRefresh() {
+        ArrayAdapter<String> spinner_adapter = new ArrayAdapter<String>(getContext(), android.R.layout.simple_spinner_dropdown_item, items);
+        exploreSpinner.setAdapter(spinner_adapter);
+        spinner_adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        exploreSpinner.setAdapter(spinner_adapter);
+        exploreSpinner.setOnItemSelectedListener(this);
+        exploreSpinner.setSelection(RELEVANCE_SORT);
+
         ivMap.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 Intent intent = new Intent(getContext(), MapActivity.class);
                 startActivity(intent);
             }
         });
+
+
         adapter = new ExploreAdapter(getContext(), restaurantList);
         rvRestaurants.setAdapter(adapter);
         LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getContext());
-
         rvRestaurants.setLayoutManager(linearLayoutManager);
         scrollListener = new EndlessRecyclerViewScrollListener(linearLayoutManager) {
             @Override
             public void onLoadMore(int page, int totalItemsCount, RecyclerView view) {
-                restaurantServer.findRestaurants(getResources().getString(R.string.yelp_api_key), restaurantListener);
-                Log.i(TAG, String.valueOf(adapter.size()));
+                if(exploreSpinner.getSelectedItemPosition() == RELEVANCE_SORT){
+                    relevanceRestaurantServer.findRestaurants(getResources().getString(R.string.yelp_api_key), restaurantListener);
+                }
+                else if(exploreSpinner.getSelectedItemPosition() == DISTANCE_SORT){
+                    distanceRestaurantServer.findRestaurants(getResources().getString(R.string.yelp_api_key), restaurantListener);
+                }
+                else if(exploreSpinner.getSelectedItemPosition() == POPULARITY_SORT){
+                    popularityRestaurantServer.findRestaurants(restaurantListener);
+                }
             }
         };
         rvRestaurants.addOnScrollListener(scrollListener);
 
-        restaurantServer.findRestaurants(getResources().getString(R.string.yelp_api_key), restaurantListener);
+        if(exploreSpinner.getSelectedItemPosition() == RELEVANCE_SORT){
+            Log.i(TAG, "RELEVANCE SORT");
+            relevanceRestaurantServer.findRestaurants(getResources().getString(R.string.yelp_api_key), restaurantListener);
+        }
+        else if(exploreSpinner.getSelectedItemPosition() == DISTANCE_SORT){
+            Log.i(TAG, "DISTANCE SORT");
+            distanceRestaurantServer.findRestaurants(getResources().getString(R.string.yelp_api_key), restaurantListener);
+        }
+        else if(exploreSpinner.getSelectedItemPosition() == POPULARITY_SORT){
+            Log.i(TAG, "POPULARITY SORT");
+            popularityRestaurantServer.findRestaurants(restaurantListener);
+        }
```

Comment:
These look like good candidates for polymorphism. 

It would be nice to have a common interface between these three. One with findRestaurants as the only api. 

so you could simple do : 

getRestaurantBySelectedPositions().findRestaurants(....)

Reply:
I'm not completely sure what getRestaurantBySelectedPositions() is supposed to do/ how it uses the common interface you describe?

Reply:
The popularity, distance, and relevence servers have common method calls that could be abstracted to an interface class that you can call the common method on. Happy to chat offline more about it :)